thorlcr/activities/trace/thtraceslave.ipp

```diff
@@ -0,0 +1,27 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _THTRACESLAVE_IPP
+#define _THTRACESLAVE_IPP
+
+#include "platform.h"
+#include "eclhelper.hpp" // tmp for IHThor..Arg interfaces.
+#include "slave.ipp"
+#include "thactivityutil.ipp"
+
+activityslaves_decl CActivityBase *createTraceSlave(CGraphElementBase *container);
```

Comment:
I have gone off the idea of creating an .ipp purely for the inclusions of the activity factory methods like this.
For more recent activities, I have added the prototype to slave.cpp directly and not used an .ipp.